src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/TrustCorporateValidator.java

```diff
@@ -0,0 +1,225 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.overseasentitiesapi.model.dao.trust.BeneficialOwnerType;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.AddressDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.trust.TrustCorporateDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.trust.TrustDataDto;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.*;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Objects;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators.setErrorMsgToLocation;
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+@Component
+public class TrustCorporateValidator {
+    public static final String PARENT_FIELD = getQualifiedFieldName(OverseasEntitySubmissionDto.TRUST_DATA,
+            TrustDataDto.CORPORATE_FIELD);
+
+    private final AddressDtoValidator addressDtoValidator;
+
+    @Autowired
+    public TrustCorporateValidator(AddressDtoValidator addressDtoValidator) {
+        this.addressDtoValidator = addressDtoValidator;
+    }
+
+    public Errors validate(List<TrustDataDto> trustDataDtoList, Errors errors, String loggingContext) {
+
+        for (TrustDataDto trustDataDto : trustDataDtoList) {
+            List<TrustCorporateDto> corporates = trustDataDto.getCorporates();
+
+            if (!CollectionUtils.isEmpty(corporates)) {
+                for (TrustCorporateDto trustCorporateDto : corporates) {
+                    validateName(trustCorporateDto.getName(), errors, loggingContext);
+
+                    String type = trustCorporateDto.getType();
+                    if(validateType(type, errors, loggingContext) && BeneficialOwnerType
+                            .findByBeneficialOwnerTypeString(type).equals(BeneficialOwnerType.INTERESTED_PERSON)) {
```

Comment:
Could this equals be the other way round perhaps? Less chance of a NullPointerEx if findByBeneficialOwnerTypeString returns null, which probably shouldn't happen, but still...

Reply:
I have had a word with Naz, who has been talking with Paul, and am happy to disregard this comment.